### PR TITLE
[API-4403] Focus Management

### PR DESCRIPTION
**Pull request by gileswells**
### Description

https://vajira.max.gov/browse/API-4403
This updates hash links and how they're handled on the auth pages specifically but also touches on the release notes pages as well. With this update the pages should properly respect hash urls on load and on in page navigation.

### Process

- [ ] Tests added or updated for change
- [ ] Docs updated
  - [ ] If you have made structural changes to the site, the [Developer Portal Content Types](https://community.max.gov/x/qUwOg) docs have been updated
  - [ ] If you have made/confirmed any decisions about user interactions and accessibility, the [Accessible Component Design](https://community.max.gov/x/xS8mgg) docs have been updated
- [ ] Accessibility concerns have been considered and addressed
  
  - [ ] If you've made any UI changes, you've run an axe check using the axe browser extension that reported no new issues

### Requested feedback

This should be tested across Chrome/FireFox/Safari as some issues were not universal.

### CODE REVIEW

File: src/components/pageContent/PageContent.tsx [hunk 1]
```diff
@@ -7,7 +7,9 @@ const focusAndScroll = (elementToFocus: HTMLElement | null): void => {
     elementToFocus.focus();
   }
 
-  window.scrollTo(0, 0);
+  if (elementToFocus?.id === 'main') {
```
Comment (mjlumetta) [inline src/components/pageContent/PageContent.tsx:10]: wondering if we should inline this `focusAndScroll` function in the effect. I don't think this conditional is necessarily bad in a separate function as a safeguard, but if we inline it, it becomes clearer that the only element we call it on is the main element.

**Review by mjlumetta — DISMISSED** [review]
yeah, it's working for me in all browsers. should we add some new integration tests? that's an AC on the ticket. anyway, I think it's all looking good from what I can tell doing manual testing.

**Comment by gileswells** [comment]
> should we add some new integration tests? that's an AC on the ticket.

I'm working on those now. I'm thinking that I might end up having to hold on the maintaining of focus parts of those tests for the auth docs pages specifically until they're enabled in prod.

**Comment by gileswells** [comment]
To get the test running you'll need to change [this line](https://github.com/department-of-veterans-affairs/developer-portal/pull/619/files#diff-a98238eb805c61b666ae4064bc2cad399d5819f34a2d49565dda5a825b9185f0R56) to remove `.skip` and also add `REACT_APP_AUTH_DOCS_V2=true` to `jest-puppeteer.config.json` under the `envVars` array.

**Review by ediiotero — APPROVED** [review]
Ran e2e test locally. Looks good

**Review by ediiotero — APPROVED** [review]
LGTM

**Pull request merged**